File: api/structs/api/task.go
package api

import "time"

type Task struct {
	Id          string    `json:"id"`
	Title       string    `json:"title"`
	Content     string    `json:"content"`
	Status      int       `json:"status"`
	Category    Category  `json:"category"`
	ScheduledAt time.Time `json:"scheduledAt"`
	CreatedAt   time.Time `json:"createdAt"`
	ModifiedAt  time.Time `json:"modifiedAt"`
}

File: app/app/src/main/java/com/solutions/note_it/data/Task.kt
package com.solutions.note_it.data

import com.google.gson.annotations.SerializedName

data class Task(
    @field:SerializedName("id") val id: String = "",
    @field:SerializedName("title") val title: String = "",
    @field:SerializedName("content") val content: String = "",
    @field:SerializedName("status") val status: Int = TaskStatus.Todo.ordinal,
    @field:SerializedName("scheduledAt") val scheduledAt: String = "",
    @field:SerializedName("category") val category: Category = Category(type = CategoryType.Task.ordinal)
)